### ISSUE TITLE
Request for the code and  detailed hyperparameter settings for the stage 2 training of the LMM-RGCL

### ISSUE THREAD

**Issue by PotatoDog1669** [JingbiaoMei/RGCL#7]
Hi! 
First of all, thank you for your solid work and for releasing the Stage 1 code. I noticed that the released Stage 1 code includes the implementation of the RGCL loss function and the feature extraction scripts in the custom directory. Based on my understanding of the Stage 2 methodology described in the paper, I attempted to implement the Stage 2 training process and tested it on the FB and HarmC datasets.

Specifically, I used extract_features.py from src.llamafactory.custom to extract features from the Stage 1-trained model and then designed train_stage2.py by referencing src.llamafactory.train.sft_classifier.train.py, executing it via run_stage2.py. However, possibly due to misunderstandings in the approach, suboptimal hyperparameter choices, or logical issues in the code, the performance improvement from Stage 2 training was only around 1%, which is lower than the results reported in your paper.

Could you kindly share the Stage 2 training code along with the detailed hyperparameter settings? I would greatly appreciate your guidance!

Thank you in advance for your time and support.

[run_stage2_FB.txt](https://github.com/user-attachments/files/20223538/run_stage2_FB.txt)
[train_stage2.txt](https://github.com/user-attachments/files/20223537/train_stage2.txt)

**Comment by JingbiaoMei** [comment]
Hi,

Thank you for your interest in the work! I am glad that you tried to implement the stage-2 yourself. I think the improvement of 1% with stage 2 is about what you can get. As shown in Table 3 of our paper, the gain from stage-2 is less than 1% on the FB, while about 1% on HarmC. Just a note, we remove the in-batch negative examples to stabilize the trainings.

**Comment by upccpu** [comment]
> Hi! First of all, thank you for your solid work and for releasing the Stage 1 code. I noticed that the released Stage 1 code includes the implementation of the RGCL loss function and the feature extraction scripts in the custom directory. Based on my understanding of the Stage 2 methodology described in the paper, I attempted to implement the Stage 2 training process and tested it on the FB and HarmC datasets.
> 
> Specifically, I used extract_features.py from src.llamafactory.custom to extract features from the Stage 1-trained model and then designed train_stage2.py by referencing src.llamafactory.train.sft_classifier.train.py, executing it via run_stage2.py. However, possibly due to misunderstandings in the approach, suboptimal hyperparameter choices, or logical issues in the code, the performance improvement from Stage 2 training was only around 1%, which is lower than the results reported in your paper.
> 
> Could you kindly share the Stage 2 training code along with the detailed hyperparameter settings? I would greatly appreciate your guidance!
> 
> Thank you in advance for your time and support.
> 
> [run_stage2_FB.txt](https://github.com/user-attachments/files/20223538/run_stage2_FB.txt) [train_stage2.txt](https://github.com/user-attachments/files/20223537/train_stage2.txt)

Hello, could you please share the RGCLArguments code from the reproduction of stage 2? Thank you very much for your contribution.

**Comment by JingbiaoMei** [comment]
> > Hi! First of all, thank you for your solid work and for releasing the Stage 1 code. I noticed that the released Stage 1 code includes the implementation of the RGCL loss function and the feature extraction scripts in the custom directory. Based on my understanding of the Stage 2 methodology described in the paper, I attempted to implement the Stage 2 training process and tested it on the FB and HarmC datasets.
> > Specifically, I used extract_features.py from src.llamafactory.custom to extract features from the Stage 1-trained model and then designed train_stage2.py by referencing src.llamafactory.train.sft_classifier.train.py, executing it via run_stage2.py. However, possibly due to misunderstandings in the approach, suboptimal hyperparameter choices, or logical issues in the code, the performance improvement from Stage 2 training was only around 1%, which is lower than the results reported in your paper.
> > Could you kindly share the Stage 2 training code along with the detailed hyperparameter settings? I would greatly appreciate your guidance!
> > Thank you in advance for your time and support.
> > [run_stage2_FB.txt](https://github.com/user-attachments/files/20223538/run_stage2_FB.txt) [train_stage2.txt](https://github.com/user-attachments/files/20223537/train_stage2.txt)
> 
> Hello, could you please share the RGCLArguments code from the reproduction of stage 2? Thank you very much for your contribution.

Hi, 

I apologize for the delay in my response, but at the moment, I am currently busy with the paper review and rebuttal. We aim to clean up the code for stage 2 and release it soon when the paper is accepted.

**Comment by JingbiaoMei** [comment]
Hi,

The full code has now been released.
